pkg/cmd/tests/tests_run.go

```diff
@@ -164,38 +258,198 @@ func (o *RunOptions) run(ctx context.Context, streams genericclioptions.IOStream
+	if err != nil {
+		return fmt.Errorf("can't create parallel spec server")
+	}
+	server.Start()
+	defer server.Close()
+
+	commonArgs := make([]string, 0, 2+len(os.Args))
+	if len(os.Args) > 1 {
+		commonArgs = append(commonArgs, os.Args[1:]...)
+	}
+	commonArgs = append(commonArgs, fmt.Sprintf("--%s=%v", parallelServerAddressFlagKey, server.Address()))
+	commonArgs = append(commonArgs, fmt.Sprintf("--%s=%v", cmdutil.FlagLogLevelKey, o.ParallelLogLevel))
+
+	type cmdEntry struct {
+		id  int
+		cmd *exec.Cmd
+		out *bytes.Buffer
+	}
+	cmdEntries := make([]*cmdEntry, 0, suiteConfig.ParallelTotal)
+	for i := 1; i <= suiteConfig.ParallelTotal; i++ {
+		suiteConfig := suiteConfig
+		suiteConfig.ParallelProcess = i
+
+		args := make([]string, 0, len(commonArgs)+1)
+		args = append(args, commonArgs...)
+		args = append(args, fmt.Sprintf("--%s=%v", parallelShardFlagKey, i))
+
+		buf := &bytes.Buffer{}
+		cmd := exec.CommandContext(ctx, os.Args[0], args...)
+		cmd.Stdout = buf
+		cmd.Stderr = buf
+		cmdEntries = append(cmdEntries, &cmdEntry{
+			id:  i,
+			out: buf,
+			cmd: cmd,
+		})
+	}
+
+	var errs []error
+	for entryIndex := range cmdEntries {
+		e := cmdEntries[entryIndex]
+		err := e.cmd.Start()
+		if err != nil {
+			errs = append(errs, fmt.Errorf("can't start command %q with args %v: %w", e.cmd.Path, e.cmd.Args, err))
+		}
+		klog.V(2).InfoS("Started process", "ID", e.id, "Process", e.cmd.String())
+		// RegisterAlive needs to be set so ginkgo worker #1 can detect all the other workers are finished
+		// and start serial tests. It needs cmd.Wait to be called first to pick up the state.
+		server.RegisterAlive(e.id, func() bool { return e.cmd.ProcessState == nil || !e.cmd.ProcessState.Exited() })
+	}
+	if len(errs) != 0 {
+		return apierrors.NewAggregate(errs)
+	}
+
+	// We need to wait for all the processes in parallel so the Alive function can read the status,
+	// which is available only after calling Wait().
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	for entryIndex := range cmdEntries {
+		e := cmdEntries[entryIndex]
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			klog.V(2).InfoS("Waiting for process", "ID", e.id, "Process", e.cmd.String())
+			err := e.cmd.Wait()
+			if err != nil {
+				errs = append(errs, fmt.Errorf("can't wait for command %q with args %q: %w", e.cmd.Path, e.cmd.Args, err))
+			}
+			klog.V(2).InfoS("Process finished", "ID", e.id, "Process", e.cmd.String(), "ExitCode", e.cmd.ProcessState.ExitCode())
+		}()
+	}
+	if len(errs) != 0 {
+		return apierrors.NewAggregate(errs)
+	}
+
+	wg.Wait()
+
+	select {
+	case <-ctx.Done():
+		return fmt.Errorf("context cancelled")
+
+	case <-server.GetSuiteDone():
+		break
+
+	default:
+		for _, e := range cmdEntries {
+			exitCode := e.cmd.ProcessState.ExitCode()
+			switch exitCode {
+			case 0, types.GINKGO_FOCUS_EXIT_CODE:
+				break
+			default:
+				klog.ErrorS(nil, "Process failed", "ID", e.id, "Process", e.cmd.String(), "Logs", e.out)
```

Comment:
e.out.String(), otherwhise you will get an array of bytes

Reply:
it doesn't output bytes
```
Logs="Error: unknown flag: --foo\n"
```
but it won't hurt :)